// angular-sortablejs/dist/src/sortablejs.directive.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var globals_1 = require("./globals");
var sortablejs_service_1 = require("./sortablejs.service");
var Sortable = require("sortablejs/Sortable.min");
var SortablejsDirective = (function () {
    function SortablejsDirective(sortablejsService, element, zone, cdr) {
        this.sortablejsService = sortablejsService;
        this.element = element;
        this.zone = zone;
        this.cdr = cdr;
        this.runInsideAngular = false;
    }
    SortablejsDirective.prototype.ngOnInit = function () {
        var _this = this;
        if (this.runInsideAngular) {
            this._sortable = Sortable.create(this.element.nativeElement, this.options);
        }
        else {
            this.zone.runOutsideAngular(function () {
                _this._sortable = Sortable.create(_this.element.nativeElement, _this.options);
            });
        }
    };
    SortablejsDirective.prototype.ngOnChanges = function (changes) {
        var _this = this;
        var optionsChange = changes['inputOptions'];
        if (optionsChange && !optionsChange.isFirstChange()) {
            var previousOptions_1 = optionsChange.previousValue;
            var currentOptions_1 = optionsChange.currentValue;
            Object.keys(currentOptions_1).forEach(function (optionName) {
                if (currentOptions_1[optionName] !== previousOptions_1[optionName]) {
                    _this._sortable.option(optionName, currentOptions_1[optionName]);
                }
            });
        }
    };
    SortablejsDirective.prototype.ngOnDestroy = function () {
        this._sortable.destroy();
    };
    Object.defineProperty(SortablejsDirective.prototype, "options", {
        get: function () {
            return Object.assign({}, globals_1.GLOBALS.options, this.inputOptions, this.overridenOptions);
        },
        enumerable: true,
        configurable: true
    });
    SortablejsDirective.prototype.proxyEvent = function (eventName, event) {
        if (this.inputOptions && this.inputOptions[eventName]) {
            this.inputOptions[eventName](event);
        }
        this.cdr.detectChanges();
    };
    Object.defineProperty(SortablejsDirective.prototype, "bindingEnabled", {
        get: function () {
            return !!this.items;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SortablejsDirective.prototype, "isItemsFormArray", {
        get: function () {
            return !!this.items.at && !!this.items.insert && !!this.items.reset;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SortablejsDirective.prototype, "overridenOptions", {
        get: function () {
            var _this = this;
            return {
                onAdd: function (event) {
                    if (_this.bindingEnabled) {
                        _this.sortablejsService.onremove = function (item) {
                            if (_this.isItemsFormArray) {
                                _this.items.insert(event.newIndex, item);
                            }
                            else {
                                _this.items.splice(event.newIndex, 0, item);
                            }
                        };
                    }
                    _this.proxyEvent('onAdd', event);
                },
                onRemove: function (event) {
                    if (_this.bindingEnabled) {
                        var item = void 0;
                        if (_this.isItemsFormArray) {
                            item = _this.items.at(event.oldIndex);
                            _this.items.removeAt(event.oldIndex);
                        }
                        else {
                            item = _this.items.splice(event.oldIndex, 1)[0];
                        }
                        _this.sortablejsService.onremove(item);
                        _this.sortablejsService.onremove = null;
                    }
                    _this.proxyEvent('onRemove', event);
                },
                onUpdate: function (event) {
                    if (_this.bindingEnabled) {
                        if (_this.isItemsFormArray) {
                            var relocated = _this.items.at(event.oldIndex);
                            _this.items.removeAt(event.oldIndex);
                            _this.items.insert(event.newIndex, relocated);
                        }
                        else {
                            _this.items.splice(event.newIndex, 0, _this.items.splice(event.oldIndex, 1)[0]);
                        }
                    }
                    _this.proxyEvent('onUpdate', event);
                }
            };
        },
        enumerable: true,
        configurable: true
    });
    return SortablejsDirective;
}());
SortablejsDirective.decorators = [
    { type: core_1.Directive, args: [{
                selector: '[sortablejs]'
            },] },
];
SortablejsDirective.ctorParameters = function () { return [
    { type: sortablejs_service_1.SortablejsService, },
    { type: core_1.ElementRef, },
    { type: core_1.NgZone, },
    { type: core_1.ChangeDetectorRef, },
]; };
SortablejsDirective.propDecorators = {
    'items': [{ type: core_1.Input, args: ['sortablejs',] },],
    'inputOptions': [{ type: core_1.Input, args: ['sortablejsOptions',] },],
    'runInsideAngular': [{ type: core_1.Input },],
};
exports.SortablejsDirective = SortablejsDirective;
