// ng2-breadcrumb/app/components/breadcrumb.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from './breadcrumb';
import * as import3 from '@angular/router';
import * as import4 from './breadcrumbService';
const styles_BreadcrumbComponent:any[] = ([] as any[]);
export const RenderType_BreadcrumbComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_BreadcrumbComponent,
  data: {}
}
);
function View_BreadcrumbComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[[
        'role',
        'button'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.navigateTo('/')) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>v.parent).context.$implicit;
    ck(v,1,0,currVal_0);
  });
}
function View_BreadcrumbComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[[
        'role',
        'button'
      ]
      ],(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.navigateTo((<any>v.parent).context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.friendlyName((<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_BreadcrumbComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.friendlyName((<any>v.parent).context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_BreadcrumbComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.friendlyName('/');
    ck(v,1,0,currVal_0);
  });
}
function View_BreadcrumbComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),16,'li',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod([
      'breadcrumb-item',
      'active'
    ]
    ),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_BreadcrumbComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_BreadcrumbComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_BreadcrumbComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_BreadcrumbComponent_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,co.useBootstrap,v.context.last);
    ck(v,1,0,currVal_0);
    const currVal_1:boolean = (!v.context.last && (v.context.$implicit == co.prefix));
    ck(v,6,0,currVal_1);
    const currVal_2:boolean = (!v.context.last && (v.context.$implicit != co.prefix));
    ck(v,9,0,currVal_2);
    const currVal_3:any = v.context.last;
    ck(v,12,0,currVal_3);
    const currVal_4:any = (v.context.last && (v.context.$implicit == co.prefix));
    ck(v,15,0,currVal_4);
  },(null as any));
}
export function View_BreadcrumbComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ul',([] as any[]),[[
        2,
        'breadcrumb',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_BreadcrumbComponent_1)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import2.BreadcrumbComponent = v.component;
    const currVal_1:any = co._urls;
    ck(v,4,0,currVal_1);
  },(ck,v) => {
    var co:import2.BreadcrumbComponent = v.component;
    const currVal_0:any = co.useBootstrap;
    ck(v,1,0,currVal_0);
  });
}
function View_BreadcrumbComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'breadcrumb',([] as any[]),(null as any),(null as any),(null as any),View_BreadcrumbComponent_0,RenderType_BreadcrumbComponent)),
    import0.ɵdid(385024,(null as any),0,import2.BreadcrumbComponent,[
      import3.Router,
      import4.BreadcrumbService
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const BreadcrumbComponentNgFactory:import0.ComponentFactory<import2.BreadcrumbComponent> = import0.ɵccf('breadcrumb',import2.BreadcrumbComponent,View_BreadcrumbComponent_Host_0,{
  useBootstrap: 'useBootstrap',
  prefix: 'prefix'
}
,{},([] as any[]));
